### PR TITLE
Added English Version 

### PR DESCRIPTION
**Pull request by NextThread**
#160 
Added English version

### CODE REVIEW

**Comment by c121914yu** [comment]
* You cann't use `useTranslation` in not-hook pages.
* For these api files, you only need to return English and use useTranslation to translate after FE receives English.

**I suggest you try to translate this file first: `client/src/pages/account/index.tsx` **

Finally, for t(""), you need to fill in the string in English, not Chinese, and the Chinese translation should be placed in `client/public/locales/zh/common.json`, the English translation  in `client/public/locales/en/common.json`.

For example: 登出

You should write: t('account.Log out'), and add translate json:

**client/public/locales/zh/common.json**
```json
{
....
account: {
   Log out: "登出"
}
...
}
```

**client/public/locales/en/common.json** 
```json
{
....
account: {
   Log out: "Log out"
}
...
}
``

**Comment by NextThread** [comment]
https://github.com/labring/FastGPT/pull/188/files#diff-a230fda8fcaeb65e8d9d6c6a7e46be5ab91f3b77f8fbd27bf477f7130b24457c


https://github.com/labring/FastGPT/pull/188/files#diff-1765c82f6dd349c4a5efe6884d44b21f0f0bdfd864fab200ced0bb2bd0f722cf

https://github.com/labring/FastGPT/pull/188/files#diff-f8c980e5d5a5fd5b2fcf72dfcb59b7041d3362cd80409341b8f19ab3851b7d25



tried to translate this file: client/src/pages/account/index.tsx

am I on the wrong path again? please confirm

**Comment by NextThread** [comment]
> https://github.com/labring/FastGPT/pull/188/files#diff-a230fda8fcaeb65e8d9d6c6a7e46be5ab91f3b77f8fbd27bf477f7130b24457c
> 
> https://github.com/labring/FastGPT/pull/188/files#diff-1765c82f6dd349c4a5efe6884d44b21f0f0bdfd864fab200ced0bb2bd0f722cf
> 
> https://github.com/labring/FastGPT/pull/188/files#diff-f8c980e5d5a5fd5b2fcf72dfcb59b7041d3362cd80409341b8f19ab3851b7d25
> 
> tried to translate this file: client/src/pages/account/index.tsx
> 
> am I on the wrong path again? please confirm

@c121914yu

**Comment by c121914yu** [comment]
> > https://github.com/labring/FastGPT/pull/188/files#diff-a230fda8fcaeb65e8d9d6c6a7e46be5ab91f3b77f8fbd27bf477f7130b24457c
> > https://github.com/labring/FastGPT/pull/188/files#diff-1765c82f6dd349c4a5efe6884d44b21f0f0bdfd864fab200ced0bb2bd0f722cf
> > https://github.com/labring/FastGPT/pull/188/files#diff-f8c980e5d5a5fd5b2fcf72dfcb59b7041d3362cd80409341b8f19ab3851b7d25
> > 尝试翻译此文件：client/src/pages/account/index.tsx
> > 难道我又走错路了？请确认
> 
> @c121914yu

yes. 

1. You need to write English in t(), not Chinese.
2. You need to add the corresponding translation in local.
3. You can install vscode's “I18n Ally” plugin to assist with translation.

I mean, you can try to modify the file: `client/src/pages/account/index`, so that I can help you to correct. This pr is too big for me to comment on one by one

**Comment by c121914yu** [comment]
> @c121914yu 
> 
> Screenshot 2023-08-27 at 12 27 06 AM
> 
> 1. Added the English inside t()
> 
> Screenshot 2023-08-27 at 12 28 33 AM
> 
> 2. Added the corresponding in common.json

yes！
We recommend that translations add prefixes by module.

example:
Confirm to log out ＞ account.Confirm to log out.

Prefixes are roughly divided: account, app, chat, kb, common

**Comment by c121914yu** [comment]
> @c121914yu 
> 
> Screenshot 2023-08-27 at 12 27 06 AM
> 
> 1. Added the English inside t()
> 
> Screenshot 2023-08-27 at 12 28 33 AM
> 
> 2. Added the corresponding in common.json

and please fetch and rebase latest code

File: client/src/api/fetch.ts [hunk 1]
```diff
@@ -60,7 +60,7 @@ export const streamFetch = ({
               });
             } else {
               return reject({
-                message: errMsg || '响应过程出现异常~',
+                message: errMsg || 'There is an exception in the response process~',
```
Comment (c121914yu) [inline client/src/api/fetch.ts:63]: The server error prompt can not be translated first, I think about the server error how to implement i18n

File: client/src/components/APIKeyModal/index.tsx [hunk 2]
```diff
@@ -63,7 +65,7 @@ const APIKeyModal = ({ onClose }: { onClose: () => void }) => {
           API 秘钥管理
         </Box>
         <Box fontSize={'sm'} color={'myGray.600'}>
-          如果你不想 API 秘钥被滥用，请勿将秘钥直接放置在前端使用~
+          {t('如果你不想 API 秘钥被滥用，请勿将秘钥直接放置在前端使用~')}
```
Comment (c121914yu) [inline client/src/components/APIKeyModal/index.tsx:68]: You need to write it in English.
Example: t('user.Open API Tips')

en: If you do not want the API key to be abused, do not place the key directly on the front end

zh: 如果你不想 API 秘钥被滥用，请勿将秘钥直接放置在前端使用~

File: client/src/components/ChatBox/ContextModal.tsx [hunk 3]
```diff
@@ -16,7 +16,7 @@ const ContextModal = ({
     <MyModal
       isOpen={true}
       onClose={onClose}
-      title={`完整对话记录(${context.length}条)`}
+      title={t(`full conversation log(${context.length}strip)`)}
```
Comment (c121914yu) [inline client/src/components/ChatBox/ContextModal.tsx:19]: If it is a translation with variables, you need to write:
t('chat.Full Context n strip', {length: context.length})

en.json:
```json
"chat":{
    "Full Context n strip": "Full context ({{length}}) strip"
}
```

zh.json
```json
"chat":{
    "Full Context n strip": "完整对话记录 ({{length}}) 条"
}
```

**Comment by c121914yu** [comment]
You need to pull the latest code and resolve conflicts first, and I recommend committing a small amount of code first. And I gave some code comment.

There are too many modified files, it is not convenient for me to merge after correction, please submit fewer files for the first time.

**Pull request closed without merging**